src/portable/dialog/da146xx/dcd_da146xx.c

```diff
@@ -430,9 +470,16 @@ static void read_rx_fifo(xfer_ctl_t *xfer, uint16_t bytes_in_fifo)
 
   if (remaining < bytes_in_fifo) receive_this_time = remaining;
 
-  uint8_t *buf = xfer->buffer + xfer->transferred + xfer->last_packet_size;
+  if (xfer->ff)
+  {
+    tu_fifo_write_n(xfer->ff, (const void *) &regs->rxd, receive_this_time);
```

Comment:
regs->rxd is hardware register not continues memory and as such can't be used in current implementation of tu_fifo_write_n

Reply:
That should work out, i extendend the fifo to posses copy modes exactly for this purpose to copy to a constant address.

Reply:
I did not noticed that it was covered.

Reply:
Within `dcd_edpt_iso_xfer()` the copy mode is set
```
if (dir == TUSB_DIR_IN)
{
  tu_fifo_set_copy_mode_write(ff, TU_FIFO_COPY_CST);
}
else
{
  tu_fifo_set_copy_mode_read(ff, TU_FIFO_COPY_CST);
}
```
But anyway, it would not solve the problem you described with the 64 bytes. The FIFO is capable of handling DMA transferes, however, its write or read pointers would need be updated in between a write/read if a wrap to the start of the FIFO buffer would occur. This needs to be handled by interrupts and as such is not feasible here.

Reply:
The copy mode stuff is to be changed to be better visible.

Reply:
I've made the mistake because of lines
```
buf8 = buf8 + (n - f->depth) * f->item_size;
...
_ff_push_n(f, buf8, n, wRel);
```
that suggests that source address (in my case register) can be modified,
and I just assumed that _ff_push_n works as it used to.
but in this case changing address of register also seems suspicious (I did not dig deep enough
to see if this will never happen).

Reply:
I remember i did `fill_tx_fifo()` extra without using read/write functions from tusb_fifo. I could do the same for the reading part but does it make sense since interrupt is too slow anyway?

Reply:
Today I tested your code (with change to access rx fifo as byte only) and It was not working for unknown reason I will let you know what come out of it tomorrow. I don't know if it was due to timing or some other small detail yet.
One more thing that may affect interrupt latency is that DA146xx chips don't have internal flash. It is all using external SPI flash in XIP mode. So when code is in cache it may work if it needs to be read it may be too slow and DMA would not be affected. Anyway I will test some more and let you know if it can work in this new shape.